v3/lints/rfc/lint_crl_valid_reason_codes.go

```diff
@@ -0,0 +1,76 @@
+package rfc
+
+/*
+ * ZLint Copyright 2023 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/util"
+)
+
+type crlHasValidReasonCode struct{}
+
+/*
+***********************************************
+RFC 5280: 5.3.1
+
+	CRL issuers are strongly
+	  encouraged to include meaningful reason codes in CRL entries;
+	  however, the reason code CRL entry extension SHOULD be absent instead
+	  of using the unspecified (0) reasonCode value.
+
+***********************************************
+*/
+func init() {
+	lint.RegisterRevocationListLint(&lint.RevocationListLint{
+		LintMetadata: lint.LintMetadata{
+			Name:          "e_crl_has_valid_reason_code",
+			Description:   "If CRL has a reason code, it must be in 5.3.1 and not be 0.",
```

Comment:
```suggestion
			Description:   "If a CRL entry has a reason code, it MUST be in RFC5280 section 5.3.1 and SHOULD be absent instead of using unspecified (0)",
```
There's a fair few changes so I don't mind if you pick and choose from the suggestion :)